nmmo/task/base_predicate.py

```diff
@@ -0,0 +1,430 @@
+# TODO: the below line will be gone after implementation
+# pylint: disable=unnecessary-pass
+from typing import List
+
+from nmmo.task.predicate import Predicate
+from nmmo.task.game_state import TeamGameState
+from nmmo.systems import skill as Skill
+from nmmo.lib.material import Material
+from nmmo.lib import utils
+
+from nmmo.entity.entity import EntityState
+
+
+class Timer(Predicate):
+  def __init__(self, num_tick: int):
+    super().__init__(num_tick)
+    self.num_tick = num_tick
+
+  def __call__(self, team_gs: TeamGameState, ent_id):
+    """True if the current tick is within the specified num_tick
+       Otherwise false.
+    """
+    return team_gs.tick <= self.num_tick
+
+
+# CHECK ME: maybe this should be the default task?
+class LiveLong(Timer):
+  def __call__(self, team_gs, ent_id):
```

Comment:
I would have expected ent_id to be passed in the constructor, but maybe I'm not understanding the use case

Reply:
I'm posting the code snippet that we discussed below.

Reply:
Predicates were revised to have the list of agents, the state of which are being evaluated.